spec/components/layout/locale_switcher_component_spec.rb

```diff
@@ -54,19 +54,37 @@
 
   context "with a few languages" do
     before do
-      allow(I18n).to receive(:available_locales).and_return(%i[en es fr])
+      allow(I18n).to receive(:available_locales).and_return(%i[ar en es fr])
     end
 
     it "renders a list of links" do
       render_inline component
 
       expect(page).to have_css "ul"
+      expect(page).to have_link "عربى", href: "/?locale=ar"
       expect(page).to have_link "English", href: "/?locale=en"
       expect(page).to have_link "Español", href: "/?locale=es"
       expect(page).to have_link "Français", href: "/?locale=fr"
       expect(page).not_to have_css "form"
     end
 
+    it "disables Turbolinks for languages with different text direction than the current one" do
+      render_inline component
+
+      expect(page).to have_link "عربى", href: "/?locale=ar" do |link|
+        link.has_xpath?("//a[@data-turbolinks='false']")
```

Comment:
I'm not that  familiar with xpath, but I've found something funny:

```ruby
         link.has_xpath?("//a[@data-turbolinks='false']") # true
         link.has_xpath?("//a[@data-turbolinks='true']") # true
```

So it looks like this assertion isn't working as expected and the test would pass even if we change the method and introduce a bug :thinking:.

I've tried a way to test this and haven't found an elegant one since it looks like we're a bit limited in component tests due to `page` being a `Capybara::Node::Simple`. This is the best I've found so far:

```ruby
      expect(page).to have_link "عربى", href: "/?locale=ar"
      expect(page).to have_css "[href='/?locale=ar'][data-turbolinks=false]"
```

Reply:
Ups, I did not notice this. Applied your suggestion which fails and succeeds as expected.